src/shared/cloudformation/cloudformation.ts

```diff
@@ -243,21 +249,34 @@ export namespace CloudFormation {
         context: { loadTemplate: typeof load } = { loadTemplate: load }
     ): Promise<Resource> {
         const template = await context.loadTemplate(templatePath)
-        const resources = template.Resources || {}
+
+        return getResourceFromTemplateResources({
+            templateResources: template.Resources,
+            handlerName
+        })
+    }
+
+    export async function getResourceFromTemplateResources(
+        params: {
+            templateResources?: TemplateResources,
+            handlerName: string,
+        }
+    ): Promise<Resource> {
+        const resources = params.templateResources || {}
```

Comment:
I feel like this should be applied to line 254, and have templateResources be a mandatory parameter. Alternately, throw an error if templateResources is undefined.

Reply:
template.yaml files aren't required to have a Resources section, so an undefined field is expected data.

Reply:
Perhaps use `templateResources: TemplateResources = {}` in `params`?